### PR TITLE
Added slack notifier

### PR DESCRIPTION
**Pull request by s1mplex**
Hey! 

My team uses Slack so I decided to extend Stoplight with another notifier. Does it look good to go?

### CODE REVIEW

File: lib/stoplight/notifier/slack.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+module Stoplight
+  module Notifier
+    # @see Base
+    class Slack < Base
+      # @return [Proc]
+      attr_reader :formatter
+
+      # @return [::Slack::Notifier]
+      attr_reader :slack_notifier
```
Comment (tfausak) [inline lib/stoplight/notifier/slack.rb]: I might prefer calling this just `slack`. What do you think?

File: lib/stoplight/notifier/slack.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+      # @param slack_notifier [::Slack::Notifier]
+      # @param formatter [Proc, nil]
+      def initialize(slack_notifier, formatter = nil)
```
Comment (tfausak) [inline lib/stoplight/notifier/slack.rb]: Does this need to accept any options? I see that slack-notifier has a default [channel and username](https://github.com/stevenosloan/slack-notifier/blob/1.2.1/readme.md#setting-defaults).

File: lib/stoplight/notifier/slack.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        @formatter = formatter || Default::FORMATTER
+      end
+
+      def notify(light, from_color, to_color, error)
+        message = formatter.call(light, from_color, to_color, error)
+
+        slack_notifier.ping(message)
+
+        message
+      end
```
Comment (tfausak) [inline lib/stoplight/notifier/slack.rb]: I would prefer removing the blank lines here, like in [`Stoplight::Notifier::HipChat#notify`](https://github.com/orgsync/stoplight/blob/v1.0.0/lib/stoplight/notifier/hip_chat.rb#L36-L40).

File: lib/stoplight/version.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # coding: utf-8
 
 module Stoplight
-  VERSION = Gem::Version.new('1.0.0')
+  VERSION = Gem::Version.new('1.1.0')
```
Comment (tfausak) [inline lib/stoplight/version.rb]: I will handle bumping the version number when I do a release, like 10eae8e6f09c5fa28ff9b5903e380ef7b4e83ef9.

File: stoplight.gemspec [hunk 5]
```diff
@@ -41,7 +41,8 @@ Gem::Specification.new do |gem|
     'rspec' => '3.3',
     'rubocop' => '0.32',
     'timecop' => '0.7',
-    'yard' => '0.8'
+    'yard' => '0.8',
+    'slack-notifier' => '1.2.1'
```
Comment (tfausak) [inline stoplight.gemspec]: This is extremely pedantic, but I like to sort dependencies by name.

**Comment by tfausak** [comment]
This looks great! Thank you for opening this pull request. I left a few line notes, but this is basically good to go. The only other thing that needs to change is [the notifiers section of the readme](https://github.com/orgsync/stoplight/blob/v1.0.0/README.md#notifiers); I will do that after merging this PR.

File: lib/stoplight/notifier/slack.rb (hunk 2 above)
Comment (s1mplex) [inline lib/stoplight/notifier/slack.rb]: I was thinking of passing an already configured `Slack::Notifier` object and then you don't need any options.

You can override those options for any `Slack::Notifier#ping` but I'm not sure it's that useful in Stoplight's usecase.

On the other hand this isn't hard to change so I can modify this PR. What do you think?

**Comment by s1mplex** [comment]
@tfausak thanks for the feedback! I updated the PR. The only thing left to do is to decide if we want to pass options to the Notifier. Waiting for you opinion :)

File: lib/stoplight/notifier/slack.rb (hunk 2 above)
Comment (tfausak) [inline lib/stoplight/notifier/slack.rb]: Ah, of course. That makes the most sense. That's what the HipChat notifier does too. Let's keep it as it is.

**Comment by tfausak** [comment]
It looks great as is! Thanks again for submitting this pull request. This is a nice feature to have.

I'll update the documentation and make a new release today.

**Comment by s1mplex** [comment]
:+1:

**Pull request merged**